src/components/FileList/FileList.js

```diff
@@ -152,30 +151,36 @@ const pathColums = []
 const columns = []
 
 /**
- * Parse Path to retrieve the complete folder structure
- * @param  {} files
- * @param  {} changes
- * @param  {} component
+ * Parse each file's path to retrieve the complete folder structure
+ * @param  {} files The files object coming from the definition
+ * @return {Object} Return a new object containing the files object modified
  */
-const parsePaths = (files, changes, component) => {
-  const changedFacets = pickBy(changes, (item, index) => index.startsWith('described.facets'))
-  return files.map((file, index) => {
-    file.facets = map(changedFacets, (glob, facets) => {
-      if (!globToRegExp(glob).test(file.path)) return
-      return facets
-        .split('described.facets.')
-        .pop()
-        .trim()
-    })
-
-    file.areFacetsDifferent =
-      file.facets.length > 0 && isEqual(Contribution.getOriginalValue(component, `files[${index}].facets`), file.facets)
-        ? ''
-        : 'facets__isEdited'
-
+const parsePaths = (files, component, preview) => {
```

Comment:
this whole method has quite high apparent complexity. Can it be simplified and/or broken into reasonably named sub methods?

Reply:
This current implementation could be used only if we maintain react-table.
Given that, we could sure evaluate if there's a way to simplify it, maybe into the way to keep some methods reusable when we will switch to ant-design/table.

Reply:
OK for now. let's re-evaluate as we get further along. My more general comment is that large complex functions are an anti-pattern IMO. Love to see them broken down into bite-sized chunks that then fit together in a logical way the folks can understand easily at all levels of detail.